src/components/Button/Button.spec.js

```diff
@@ -1,43 +1,113 @@
-import {Button} from './index';
-import TestRenderer from 'react-test-renderer';
+import {Button, ButtonColors, ButtonSizes, ButtonVariants} from './index';
+import Love from '~/src/asset/Love.svg';
 
 describe('Button', () => {
-    it('should work', () => {
-        const wrapper = shallow(<Button onClick={() => {}}/>);
-        expect(wrapper.html());
+    it('should render', () => {
+        const button = shallow(<Button onClick={() => {}}/>);
+        expect(button.html()).toEqual('<button class="button size_default variant_default color_default icon-button" type="button" disabled="false" onClick="[onClick]"></button>');
     });
 
-    it('contains label if specified', () => {
-        const wrapper = shallow(<Button label="Button" onClick={() => {}}/>);
-        expect(wrapper.html()).toContain('Button');
+    it('should have the specified label', () => {
+        const label = 'Button Toto';
+        const button = shallow(<Button label={label} onClick={() => {}}/>);
+        expect(button.html()).toContain(label);
     });
 
     it('should display the icon', () => {
-        const Icon = () => <svg/>;
-        const testRenderer = TestRenderer.create(<Button icon={<Icon/>} onClick={() => {}}/>);
-        expect(testRenderer.root.findByType(Icon));
+        const button = shallow(<Button icon={<Love/>} onClick={() => {}}/>);
+        expect(button.querySelector('Svg').exists()).toBeTruthy();
     });
 
-    it('should use the variant ghost', () => {
-        const wrapper = shallow(<Button variant="ghost" onClick={() => {}}/>);
-        expect(wrapper.html()).toContain('button ghost inherit');
+    it('should have the specified label and an icon', () => {
+        const label = 'Button Toto';
+        const button = shallow(<Button label={label} icon={<Love/>} onClick={() => {}}/>);
+        expect(button.html()).toContain(label);
+        expect(button.querySelector('Svg').exists()).toBeTruthy();
     });
 
-    it('should use the color reverse', () => {
-        const wrapper = shallow(<Button color="reverse" onClick={() => {}}/>);
-        expect(wrapper.html()).toContain('button default reverse');
+    it('should use the variant default', () => {
+        const button1 = shallow(<Button onClick={() => {}}/>);
+        expect(button1.querySelector('.variant_default').exists()).toBeTruthy();
+        expect(button1.querySelector('.variant_outlined').exists()).toBeFalsy();
+        expect(button1.querySelector('.variant_ghost').exists()).toBeFalsy();
+
+        const button2 = shallow(<Button variant="default" onClick={() => {}}/>);
+        expect(button2.querySelector('.variant_default').exists()).toBeTruthy();
+        expect(button2.querySelector('.variant_outlined').exists()).toBeFalsy();
+        expect(button2.querySelector('.variant_ghost').exists()).toBeFalsy();
+    });
+
+    it('should use the specified variant', () => {
+        // eslint-disable-next-line array-callback-return
+        ButtonVariants.map(variant => {
```

Comment:
You could/should use <array>.forEach(...) instead of <array>.map(...). You would not need to disable the eslint rule then.

Reply:
good point